### PR TITLE
Add C implementation for `@stdlib/math/base/special/cosh`

### LINKED ISSUE #722: [RFC]: Add C implementation for `@stdlib/math/base/special/cosh`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/cosh`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/cosh).

```c
double stdlib_base_cosh( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #722.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a final review.

File: lib/node_modules/@stdlib/math/base/special/cosh/README.md [hunk 1]
```diff
@@ -77,6 +77,95 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+<!-- C API usage examples. -->
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/cosh.h"
+#include <stdlib.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cosh/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/cosh/examples/c/example.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/cosh.h"
+#include <stdlib.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cosh/examples/c/example.c]:
```suggestion
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**